### PR TITLE
Remove the networkExporter option from the helm chart

### PR DESCRIPTION
**Pull request by dmitryax**
Network explorer is no longer part of this helm chart and should be installed separately. See https://github.com/signalfx/splunk-otel-collector-chart/blob/main/docs/advanced-configuration.md#using-opentelemetry-ebpf-helm-chart-with-splunk-opentelemetry-collector-for-kubernetes for more details.

The helm chart installation will fail if `networkExporter.enabled=true` 
```
Error: UPGRADE FAILED: execution error at (splunk-otel-collector/templates/NOTES.txt:86:3): Network exporter is not part of this helm chart anymore. Please follow https://github.com/signalfx/splunk-otel-collector-chart/blob/main/UPGRADING.md#0870-to-0880 to replace it with the upstream OpenTelemetry eBPF helm chart.
```

and will print a warning if networkExporter is used in any other way
```
[WARNING] `networkExplorer` option is no longer supported. Please remove it from your custom values.yaml.
```

### CODE REVIEW

**Review by jvoravong — APPROVED** [review]

**Comment by jvoravong** [comment]
We might want to check if there are any Network Explorer metrics being tested in the v2 functional tests. LGTM overall.

**Review by samiura — APPROVED** [review]

File: helm-charts/splunk-otel-collector/templates/NOTES.txt [hunk 1]
```diff
@@ -85,7 +82,9 @@ Splunk Network Explorer is installed and configured.
   - Some libraries may be enabled by default. For current status, see: https://github.com/open-telemetry/opentelemetry-operator#controlling-instrumentation-capabilities
   - Splunk provides best-effort support for native OpenTelemetry libraries, and full support for Splunk library distributions. For used libraries, refer to the values.yaml under "operator.instrumentation.spec".
 {{- end }}
-{{- if .Values.networkExplorer.enabled }}
-[WARNING] `networkExplorer` option is deprecated and will be removed soon.
-          Follow https://github.com/signalfx/splunk-otel-collector-chart/blob/main/UPGRADING.md#0870-to-0880 to replace it with the upstream OpenTelemetry eBPF helm chart.
+{{- if and (not (eq (toString .Values.networkExplorer) "<nil>")) .Values.networkExplorer.enabled }}
+{{ fail "Network exporter is not part of this helm chart anymore. Please follow https://github.com/signalfx/splunk-otel-collector-chart/blob/main/UPGRADING.md#0870-to-0880 to replace it with the upstream OpenTelemetry eBPF helm chart." }}
```
Comment (aunshc) [inline helm-charts/splunk-otel-collector/templates/NOTES.txt:86]:
```suggestion
{{ fail "Network explorer is not part of this helm chart anymore. Please follow https://github.com/signalfx/splunk-otel-collector-chart/blob/main/UPGRADING.md#0870-to-0880 to replace it with the upstream OpenTelemetry eBPF helm chart." }}
```

**Review by aunshc — CHANGES_REQUESTED** [review]

File: helm-charts/splunk-otel-collector/templates/NOTES.txt (hunk 1 above)
Comment (dmitryax) [reply helm-charts/splunk-otel-collector/templates/NOTES.txt]: Thanks! fixed

**Comment by dmitryax** [comment]
> We might want to check if there are any Network Explorer metrics being tested in the v2 functional tests.

I haven't found any

**Comment by dmitryax** [comment]
Looks like the tests are flaky

**Pull request merged**